gradoop-examples/src/main/java/org/gradoop/examples/aggregation/AggregationExample.java

```diff
@@ -0,0 +1,137 @@
+import org.gradoop.flink.util.GradoopFlinkConfig;
+
+/**
+ * A self contained example on how to use the aggregate operator on Gradoop's LogicalGraph class.
+ *
+ * The example uses the graph in dev-support/social-network.pdf
+ */
+public class AggregationExample {
+
+  /**
+   * Path to the example data graph
+   */
+  private static final String EXAMPLE_DATA_FILE =
+    AggregationExample.class.getResource("/data/gdl/sna.gdl").getFile();
+  /**
+   * Property key 'name'
+   */
+  private static final String PROPERTY_KEY_NAME = "name";
+  /**
+   * Property key 'list_of_names'
+   */
+  private static final String PROPERTY_KEY_LIST_OF_NAMES = "list_of_names";
+  /**
+   * Property key 'birthday'
+   */
+  private static final String PROPERTY_KEY_BIRTHDAY = "birthday";
+  /**
+   * Property key 'mean_age'
+   */
+  private static final String PROPERTY_KEY_MEAN_AGE = "mean_age";
+  /**
+   * Property key 'person'
+   */
+  private static final String LABEL_PERSON = "Person";
+
+  /**
+   * Runs the program on the example data graph.
+   *
+   * The example provides an overview over the usage of the aggregate() method.
+   * It both showcases the application of aggregation functions that are already provided by
+   * Gradoop, as well as the definition of custom ones.
+   * Documentation for all available aggregation functions as well as a detailed description of the
+   * aggregate method can be found in the projects wiki.
+   *
+   * @see <a href="https://github.com/dbs-leipzig/gradoop/wiki/Unary-Logical-Graph-Operators">
+   * Gradoop Wiki</a>
+   *
+   * Using the social network graph in the resources directory, the program will:
+   * 1. extract a subgraph only containing vertices which are labeled "person"
+   * 2. concatenate the values of the vertex property "name" of each vertex in the subgraph
+   * 3. count the amount of vertices in the subgraph (aka the amount of persons)
+   * 4. sum up the values of the vertex property "birthday"
+   * 5. add the property "meanAge" to the graph head using a graph head transformation
+   *
+   * @param args arguments
+   */
+  public static void main(String[] args) throws Exception {
+    // init execution environment
+    ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+    // create loader
+    FlinkAsciiGraphLoader loader = new FlinkAsciiGraphLoader(GradoopFlinkConfig.createConfig(env));
+
+    // load data
+    loader.initDatabaseFromFile(EXAMPLE_DATA_FILE);
+
+    // get LogicalGraph representation of the social network graph
+    LogicalGraph networkGraph = loader.getDatabase().getDatabaseGraph();
+
+    // execute aggregations and graph-head transformations
+    LogicalGraph result = networkGraph
+      // extract subgraph with edges and vertices which are of interest
+      .vertexInducedSubgraph(new ByLabel<>(LABEL_PERSON))
+      // apply custom VertexAggregateFunctions
+      .aggregate(new VertexAggregateFunction() {
```

Comment:
Instead of a anonymous class i prefer a class (within package functions) which implements the interface `VertexAggregateFunction`. That should enhance the readability of this example.